app/start_handlers.py

```diff
@@ -47,41 +47,42 @@ async def post(self, request):
 
         data = await request.post()
 
-        if (not data.get('uac')) or (data.get('uac') == ''):
+        uac = data.get('uac')
+        if not uac:
             return self._display_missing_uac_error(request, display_region)
 
+        combined_uac = uac.upper().replace(' ', '')
+        if len(combined_uac) < UAC_LENGTH:
+            return self._display_malformed_uac_message(request, display_region, BAD_CODE_LENGTH_MSG,
+                                                       BAD_CODE_LENGTH_MSG_CY)
+
         try:
-            request['uac_hash'] = self._uac_hash(data.get('uac'))
+            request['uac_hash'] = self._uac_hash(combined_uac)
         except TypeError:
-            return self._display_malformed_uac_message(request, display_region)
+            return self._display_malformed_uac_message(request, display_region, INVALID_CODE_MSG, INVALID_CODE_MSG_CY)
 
         token = await EqLaunch.get_token(request, display_region, request.app)
         EqLaunch.call_eq(request.app['EQ_URL'], token)
 
     @staticmethod
-    def _display_malformed_uac_message(request, display_region):
+    def _display_malformed_uac_message(request, display_region, en_code, cy_code):
```

Comment:
Small point, but perhaps worth renaming `en_code` and `cy_code` to make it clear they are the messages to display.